src/proof/fischlin.rs

```diff
@@ -0,0 +1,357 @@
+// Author: dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+
+use crate::language::WitnessSpaceValue;
+use crate::{language, Error, Result};
+use crypto_bigint::rand_core::CryptoRngCore;
+use crypto_bigint::Concat;
+use group::{ComputationalSecuritySizedNumber, GroupElement};
+use merlin::Transcript;
+use proof::TranscriptProtocol;
+use serde::{Deserialize, Serialize};
+use std::array;
+
+/// A Universally Composable (UC) Maurer Zero-Knowledge Proof via Fischlin's transform.
+/// Implements [Chen and Lindell (2024)](https://eprint.iacr.org/2024/526.pdf).
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct Proof<
+    // Number of parallel repetitions required to get a negligible soundness error.
+    const REPETITIONS: usize,
+    // The language we are proving
+    Language: language::Language<REPETITIONS>,
+    // A struct used by the protocol using this proof,
+    // used to provide extra necessary context that will parameterize the proof (and thus verifier
+    // code) and be inserted to the Fiat-Shamir transcript
+    ProtocolContext: Clone,
+> {
+    pub maurer_proof: super::Proof<REPETITIONS, Language, ProtocolContext>,
```

Comment:
Yes proof should have REPETITIONS, Language I think shouldn't.

Reply:
Maybe that's a better design, but all my code already works this way - we can create an issue for that in the future